packages/contractkit/src/utils/wallet.ts

```diff
@@ -0,0 +1,115 @@
+  signTransaction: (txParams: CeloTx) => Promise<EncodedTransaction>
+  signTypedData: (address: string, typedData: EIP712TypedData) => Promise<string>
+  signPersonalMessage: (address: string, data: string) => Promise<string>
+}
+
+export function normalizeKey(key: string) {
+  return ensureLeading0x(key).toLowerCase()
+}
+
+export class Wallet implements IWallet {
+  // Account addresses are hex-encoded, lower case alphabets
+  private readonly accountAddressToPrivateKey = new Map<string, string>()
+
+  addAccount(privateKey: string) {
+    // Prefix 0x here or else the signed transaction produces dramatically different signer!!!
+    privateKey = normalizeKey(privateKey)
+    const accountAddress = normalizeKey(privateKeyToAddress(privateKey))
+    if (this.accountAddressToPrivateKey.has(accountAddress)) {
+      return
+    }
+    this.accountAddressToPrivateKey.set(accountAddress, privateKey)
+  }
+
+  getAccounts(): string[] {
+    return Array.from(this.accountAddressToPrivateKey.keys())
+  }
+
+  hasAccount(address: string) {
+    if (address) {
+      return this.accountAddressToPrivateKey.has(normalizeKey(address))
+    } else {
+      return false
+    }
+  }
+
+  async signTransaction(txParams: CeloTx): Promise<EncodedTransaction> {
+    return signTransaction(txParams, this.getPrivateKeyFor(txParams.from!))
+  }
+
+  // Original code taken from
+  // https://github.com/0xProject/0x-monorepo/blob/78c704e3d/packages/subproviders/src/subproviders/private_key_wallet.ts
+  /**
+   * Sign a personal Ethereum signed message.
+   * The address must be provided it must match the address calculated from the private key.
+   * @param address Address of the account to sign with
+   * @param data Hex string message to sign
+   * @return Signature hex string (order: rsv)
+   */
+  async signPersonalMessage(address: string, data: string): Promise<string> {
+    if (!isHexString(data)) {
+      throw Error('Expected data has to be an Hex String ')
+    }
+    // ecsign needs a privateKey without 0x
+    const pk = trimLeading0x(this.getPrivateKeyFor(address))
+    const pkBuffer = Buffer.from(pk, 'hex')
+
+    const dataBuff = ethUtil.toBuffer(data)
+    const msgHashBuff = ethUtil.hashPersonalMessage(dataBuff)
+
+    const sig = ethUtil.ecsign(msgHashBuff, pkBuffer)
+    const rpcSig = ethUtil.toRpcSig(sig.v, sig.r, sig.s)
+    return rpcSig
+  }
+
+  // Original code taken from
+  // https://github.com/0xProject/0x-monorepo/blob/78c704e3d/packages/subproviders/src/subproviders/private_key_wallet.ts
+  /**
+   * Sign an EIP712 Typed Data message. The signing address will be calculated from the private key.
+   * The address must be provided it must match the address calculated from the private key.
+   * @param address Address of the account to sign with
+   * @param data the typed data object
+   * @return Signature hex string (order: rsv)
+   */
+  async signTypedData(address: string, typedData: EIP712TypedData): Promise<string> {
```

Comment:
nit: dedup with `signPersonalMessage`
this code block also looks familiar, see if there is anything in utils which does the same

Reply:
mmm.. for EIP712 i'm pretty sure we don't.. but idk.

Reply:
No, we don't have the version for the EIP712
I tried not to change these function because are being copied from the 0x repo. I know that I could pass a function parameter and unify all, but probably won't be clear for the person following the link to the place that was extracted